frame/staking/src/lib.rs

```diff
@@ -636,6 +708,24 @@ pub trait Trait: frame_system::Trait {
 
 	/// The NPoS reward curve to use.
 	type RewardCurve: Get<&'static PiecewiseLinear<'static>>;
+
+	/// Something that can predict the next session change.
+	type NextSessionChange: EstimateNextSessionChange<Self::BlockNumber>;
+
+	/// How many blocks ahead of the era do we try to run the phragmen offchain? Setting this to
+	/// zero will disable the offchain compute and only on-chain seq-phragmen will be used.
+	type ElectionLookahead: Get<Self::BlockNumber>;
+
+	/// The overarching call type.
+	type Call: From<Call<Self>> + Clone;
+
+	/// A transaction submitter.
+	type SubmitTransaction: SubmitUnsignedTransaction<Self, <Self as Trait>::Call>;
+
+	/// Key type used to sign and verify transaction.
+	// TODO: this cen be fetched from `SubmitTransaction` as well, but for now both me and @todr
+	// gave up.
+	type KeyType: RuntimeAppPublic + Member + Parameter;
```

Comment:
Maybe we will still re-think this but for the time being, and if we only want to support unsigned from validators, then this is waaaaay easier.

Reply:
I suppose you mean unsigned by account keys, but still signed by some key the validator controls, yes?

Reply:
Yes; the entire payload must be signed by this key type, and is checked upon arrival. You can see a glimpse of it in the `ValudateUnsigned` at the bottom of the staking source file.